client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -115,13 +115,11 @@ public SwarmClient(Options options) {
                         new String(
                                 Files.readAllBytes(Paths.get(options.labelsFile)),
                                 StandardCharsets.UTF_8);
-                options.labels.addAll(Arrays.asList(labels.split(" ")));
+                options.labels.addAll(Arrays.asList(labels.trim().split("\\s+")));
                 logger.info("Labels found in file: " + labels);
                 logger.info(
                         "Effective label list: "
-                                + Arrays.toString(options.labels.toArray())
-                                        .replaceAll("\n", "")
-                                        .replaceAll("\r", ""));
+                                + Arrays.toString(options.labels.toArray()));
```

Comment:
[nit] I think it would be simpler to call `List#toString()` rather than converting the list to an array and then calling `Arrays#toString`, but I do not feel strongly about this.

Reply:
Me neither, but just 'cause I didn't realize/think I could do that :)

Also went for minimal changes regarding this PR's cause. Maybe it was not List but Array some releases ago?